cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/NativeProvisioner.java

```diff
@@ -38,6 +40,18 @@
 
   public static final ProvisionerSpecification SPEC = new ProvisionerSpecification(
     Profile.NATIVE_NAME, "Native", "Runs programs on the CDAP master cluster. Does not provision any resources.");
+  public static final Set<String> TRANSACTIONAL_DATASETS =
+    Collections.unmodifiableSet(Stream.of("table",
+                                          "keyValueTable",
```

Comment:
can you create constants for these values and use them here as well as the relevant dataset modules? I think it makes sense to add a TYPE constant in the relevant dataset interface. For example, Table.TYPE, PartitionedFileSet.TYPE. That way plugin authors can use them too.